### ISSUE TITLE
Frog mblem crash: folia::ValueError: attribute 'class' is required for lemma (empty class passed)

### ISSUE THREAD

**Issue by proycon** [LanguageMachines/frog#38]
mblem gets an empty class and crashes, input document in ``/vol/tensusers/proycon/vrie047wond01_01.tok.translated.folia.xml``

```
frog-:Frogging froginput/vrie047wond01_01.tok.translated.folia.xml
frog-tok-:ucto: --filter=NO is automatically set. inputclass equals outputclass!
frog-mblem:attribute 'class' is required for lemma addLemma failed.
terminate called after throwing an instance of 'folia::ValueError'
  what():  attribute 'class' is required for lemma`
Program received signal SIGABRT, Aborted.
[Switching to Thread 0x2aaab3934700 (LWP 142807)]
0x00002aaaac941c37 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
56      ../nptl/sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  0x00002aaaac941c37 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00002aaaac945028 in __GI_abort () at abort.c:89
#2  0x00002aaaac233535 in __gnu_cxx::__verbose_terminate_handler() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00002aaaac2316d6 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00002aaaac230799 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00002aaaac23134a in __gxx_personality_v0 () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00002aaaac4e6fd3 in ?? () from /lib/x86_64-linux-gnu/libgcc_s.so.1
#7  0x00002aaaac4e735b in _Unwind_RaiseException () from /lib/x86_64-linux-gnu/libgcc_s.so.1
#8  0x00002aaaac4e75dd in _Unwind_Resume_or_Rethrow () from /lib/x86_64-linux-gnu/libgcc_s.so.1
#9  0x00002aaaac231969 in __cxa_rethrow () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#10 0x00002aaaaad7c2c8 in Mblem::addLemma (this=this@entry=0x65b3e0, word=word@entry=0x64236a60, cls="") at mblem_mod.cxx:219
#11 0x00002aaaaad7c771 in Mblem::Classify (this=0x65b3e0, sword=<optimized out>) at mblem_mod.cxx:339
#12 0x00002aaaaad4a0b1 in FrogAPI::TestSentence () at FrogAPI.cxx:470
#13 0x00002aaaad8b034a in ?? () from /usr/lib/x86_64-linux-gnu/libgomp.so.1
#14 0x00002aaaac6f5184 in start_thread (arg=0x2aaab3934700) at pthread_create.c:312
#15 0x00002aaaaca08ffd in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:11
```

**Comment by kosloot** [comment]
Well, this is 'interesting'
In a few special cases the lemmatizer uses tokenizer results to get a lemma.

For instance for the Uppercased word : "In't"
The tokenizer identifies this as a WORD-WITHSUFFIX, and knows how to strip the suffix to get a lemma.
But in this case there are 2 texts associated with the word:
```xml
          <w xml:id="vrie047wond01_01.TEI.2.text.body.div.p.note.5800.s.1.w.1" class="WORD-WITHSUFFIX">
            <t>In't</t>
            <t class="contemporary">In</t>
          </w>```
AND: we are processing the textclass 'contemporary'
The 'stripping suffix' rule bluntly strips the last 2 characters from "In", leaving an empty word.
So that is wrong and can be improved on.

But there is a fundamental problem: the tokenization was done on the 'current' textclass, but this information is not available. Or we must agree that is ALWAYS on the the 'current' class only. But is NOT what we want.
Should words also carry a textclass? which in this case would mean that  we don't have words at al!?
Frog should then extract 'contemporary' sentences from the FoLiA and tokenize those. 
A lot of issues arise from here...

Help/thoughts welcome

**Comment by proycon** [comment]
The hack worked fine for my batch, the more fundamental problem is addressed in proycon/folia#32

**Comment by kosloot** [comment]
As a note: This is now solved more fundamentally by implementing the textclass attribute for words
